src/irmin/s.ml

```diff
@@ -558,6 +558,19 @@ module type COMMIT_HISTORY = sig
   (** Same as {{!Private.Node.GRAPH.closure} GRAPH.closure} but for the history
       graph. *)
 
+  val iter :
+    [> `Read ] t ->
+    min:node list ->
+    max:node list ->
+    ?commit:(commit -> unit Lwt.t) ->
+    ?edge:(node -> node -> unit Lwt.t) ->
+    ?skip:(node -> bool Lwt.t) ->
+    ?rev:bool ->
+    unit ->
```

Comment:
Why is this last `unit` argument required?

Reply:
Presumably because of the optional arguments.

Reply:
Perhaps they could be moved before the required ones? That would look cleaner to me, but I don't really mind.

Reply:
Just to be clear: the relative ordering of named and optional arguments doesn't matter here because of argument re-ordering:

```ocaml
let f ?a:_ ~b:_ : unit = ()
let () = f ~b:0
(*       ^^^^^^
 * Error: This expression has type ?a:'a -> unit
 *         but an expression was expected of type unit *)
```

but moving everything in front of the only positional argument would work.